mllib/src/main/scala/org/apache/spark/mllib/regression/StreamingDecay.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.mllib.regression
+
+import org.apache.spark.Logging
+
+trait StreamingDecay {
+  def getDiscount(numNewDataPoints: Long): Double
+}
+
+private[mllib] trait StreamingDecaySetter[T <: StreamingDecaySetter[T]] extends Logging {
+  self: T =>
+  var decayFactor: Double = 0
+  var timeUnit: String = StreamingDecay.BATCHES
```

Comment:
Oh didn't see that on L22, yeah I agree (thanks for catching :+1:)!

What I meant here is that since you provide getters and setters for `decayFactor` and `timeUnit` in the trait, perhaps we could make the actual `var`s -> `private var`s to force users to use the getters / setters (and provide an additional level of indirection separating interface from implementation)